### PR TITLE
add admin view for searching through all users interactively

### PR DESCRIPTION
**Pull request by ViRu-ThE-ViRuS**
#201  : added admin view for listing all users

**discussion:**
- filtering through email needs to incorporate multiple email addresses?
- is this to replace `User Search` admin page
- incorporate more filters

### CODE REVIEW

**Comment by matthew-li** [comment]
@ViRu-ThE-ViRuS

> * filtering through email needs to incorporate multiple email addresses?

Yes.

> * is this to replace `User Search` admin page

Let's create a new view. We can then evaluate whether we can hide User Search later.

> * incorporate more filters

A minor one that comes to mind is `middle_name`. Did you have anything else in mind?

**Comment by ViRu-ThE-ViRuS** [comment]
@matthew-li ready for review!

File: coldfront/core/user/forms.py [hunk 1]
```diff
@@ -28,6 +29,20 @@ class UserSearchForm(forms.Form):
...
+    MIDDLE_NAME = 'Middle Name'
+    LAST_NAME = 'Last Name'
+    USERNAME = 'Cluster Username'
+    EMAIL = 'Email'
+
+    first_name = forms.CharField(label=FIRST_NAME, max_length=100, required=False)
+    middle_name = forms.CharField(label=MIDDLE_NAME, max_length=100, required=False)
+    last_name = forms.CharField(label=LAST_NAME, max_length=100, required=False)
+    username = forms.CharField(label=USERNAME, max_length=100, required=False)
+    email = forms.EmailField(label=EMAIL, max_length=100, required=False)
```
Comment (matthew-li) [inline coldfront/core/user/forms.py:43]: Thanks for adding support for other email addresses. Let's add a `help_text` field here with the following text:

"You may use this field to find the user that an email address belongs to, even if it is not a primary address."

File: coldfront/core/user/views.py [hunk 2]
```diff
@@ -309,6 +310,104 @@ def test_func(self):
...
 
 
+class UserSearchAll(LoginRequiredMixin, ListView):
+    model = User
+    template_name = 'user/user_list.html'
+    context_object_name = 'user_list'
+    paginate_by = 25
+
+    def test_func(self):
+        return self.request.user.is_staff
```
Comment (matthew-li) [inline coldfront/core/user/views.py:320]: This should be `is_superuser`. We will open up permissions later once we've reviewed who should get access to what.

File: coldfront/templates/common/navbar_admin.html [hunk 3]
```diff
@@ -3,6 +3,7 @@
   <div class="dropdown-menu">
     <a class="dropdown-item" href="/admin">MyBRC Administration</a>
     <a id="navbar-user-search" class="dropdown-item" href="{% url 'user-search-home' %}">User Search</a>
+    <a class="dropdown-item" href="{% url 'user-search-all' %}">All Users</a>
```
Comment (matthew-li) [inline coldfront/templates/common/navbar_admin.html:6]: FYI (no action needed): If we wanted this to be available to non-admin staff (which we don't), we'd also want to add this to `navbar_nonadmin_staff.html`.

**Review by matthew-li — CHANGES_REQUESTED** [review]
This looks and works great! Thanks for getting it in.

I had a few very minor comments.

**Comment by ViRu-ThE-ViRuS** [comment]
@matthew-li @kmuriki ready for review

**Review by matthew-li — APPROVED** [review]

**Pull request merged**